src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetRandomCommand.cs

```diff
@@ -275,7 +275,6 @@ private double ConvertToDouble(object o, double defaultIfNull)
         /// List from which random elements are chosen.
         /// </summary>
         [Parameter(ParameterSetName = RandomListItemParameterSet, ValueFromPipeline = true, Position = 0, Mandatory = true)]
-        [ValidateNotNullOrEmpty]
```

Comment:
Without [AllowNull] we still have different behavior for:
 Get-Random -InputObject @('zzz', 'aaa', $null)
 Get-Random @('zzz', 'aaa', $null)

Reply:
Good point, @iSazonov: Instead of removing the attribute, it should be replaced with `[System.Management.Automation.AllowNullAttribute]` so as to allow `$null` values, which then requires additional changes: 

* declare static helper field 
`private static object[] _nullInArray = new object[] { null };`

* In `ProcessRecord()` replace `foreach (object item in InputObject)` with 
`foreach (object item in InputObject ?? _nullInArray)`

(As before, passing a _scalar_ positionally - including `$null` - will continue to bind to `-Maximum`).

Reply:
We shouldn't break existing `Maximum` behavior.  I'll make the proposed changes to have them consistent.